src/main/scala/sparkz/core/api/http/ApiDirectives.scala

```diff
@@ -18,4 +20,27 @@ trait ApiDirectives extends CorsHandler with SparkzEncoding {
       else reject(AuthorizationFailedRejection)
   }
 
+  val DEFAULT_USER: String = "user"
+
+  lazy val withBasicAuth: AuthenticationDirective[String] = authenticateBasic(realm = "secure api", basicAuthentication)
+
+  def basicAuthentication(credentials: Credentials): Option[String] = {
+    settings.apiKeyHash.isEmpty match {
+      case true => Some(DEFAULT_USER)
+      case false =>
+            credentials match {
+              case p @ Credentials.Provided(id) if p.provideVerify(verifyApiKey) => Some(id)
+              case _ => None
+            }
+    }
+  }
+
+  def verifyApiKey(apiKey: String): Boolean = {
+    val apiKeyHash = settings.apiKeyHash.fold("")(key => key)
+    if (apiKeyHash.equals(""))
+      false
+    else
```

Comment:
```
settings.apiKeyHash match {
  case Some(apiKeyHash) => BCrypt.checkpw(apiKey, apiKeyHash)
  case None => false
}
```

Reply:
The problem is that if the apiKeyHash is "" the BCrypt raise a  `StringIndexOutOfBoundsException ` instead of rejects the request

Reply:
Done